.github/workflows/cpp-linter.yml

```diff
@@ -0,0 +1,36 @@
+name: cpp-linter
+permissions: read-all
+
+on:
+  pull_request:
+    branches:
+      - master
+    paths-ignore: "tests/**"
+
+jobs:
+  cpp-linter:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+#       - name: cmake
+#         run: CXX=clang++-12 cmake -G "Unix Makefiles" -DCMAKE_BUILD_TYPE=Release .
+      - uses: shenxianpeng/cpp-linter-action@master
```

Comment:
There are options for changed lines or changed files only. We should have those on, unless all our files lints are already lint free. It looks like there are some lints in bfbs_gen_lua so consider fixing those (and all others) or add these flags so we can fix the lints incrementally. 

https://github.com/shenxianpeng/cpp-linter-action

Reply:
Yeah, there is an option for lines changed only, so I will probably turn that on. I just wanted to see how it would show the errors.

Reply:
Added lines changed only.